### PR TITLE
Add releasing tools

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

Add some releasing tools to help simplify the releasing process.
The scripts are copied from [Apache Flink](https://github.com/apache/flink/tree/master/tools/releasing) project and modified as needed.

## Brief change log

- Add create_release_binary.sh
- Add create_source_release.sh
- Add deploy_staging_jars.sh
- Change the compression method of AMS from zip to tar.gz

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by baiyangtx — APPROVED** [review]
LGTM, thanks for your contributions

File: docs/admin-guides/deployment.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ You can choose to download the stable release package from [download page](../..
 ## Download the distribution
 
 All released package can be downloaded from [download page](../../../download/).
-You can download amoro-x.y.z-bin.zip (x.y.z is the release number), and you can also download the runtime packages for each engine version according to the engine you are using.
+You can download apache-amoro-x.y.z-bin.tar.gz (x.y.z is the release number), and you can also download the runtime packages for each engine version according to the engine you are using.
 Unzip it to create the amoro-x.y.z directory in the same directory, and then go to the amoro-x.y.z directory.
```
Comment (tcodehuber) [inline docs/admin-guides/deployment.md:28]: apache-amoro-x.y.z-bin directory in the same directory, and then go to the apache-amoro-x.y.z-bin directory.

**Review by tcodehuber — COMMENTED** [review]
Thanks for your great work, I left one comment, PTAL.

File: docs/admin-guides/deployment.md (hunk 1 above)
Comment (zhoujinsong) [reply docs/admin-guides/deployment.md:28]: The directory of the unzip result remains `amoro-x.y.z`.

**Pull request merged**